contracts/remarkables/src/contract.rs

```diff
@@ -0,0 +1,922 @@
+        token_id: token_id.clone(),
+        owner: info.sender.clone().into(),
+        token_uri: Some(remarkables_uri.clone()),
+        extension: Empty {},
+    });
+    let wasm_execute_mint_msg = wasm_execute(CW721_ADDRESS.load(deps.storage)?, &mint_msg, vec![])?;
+    Ok(Response::new()
+        .add_attribute(ATTRIBUTE_ACTION, ACTION_MINT_TO)
+        .add_attribute(ATTRIBUTE_SENDER, &info.sender)
+        .add_attribute(ATTRIBUTE_RARITY_LEVEL, rarity_level.to_string())
+        .add_attribute(ATTRIBUTE_TOKEN_ID, token_id.to_string())
+        .add_attribute(ATTRIBUTE_RECIPIENT, &info.sender)
+        .add_attribute(ATTRIBUTE_TOKEN_URI, remarkables_uri)
+        .add_message(wasm_execute_mint_msg))
+}
+
+// Returns the token id as "<post-id>-<rarity-level>"
+pub fn convert_post_id_to_token_id(post_id: u64, rarity_level: u32) -> String {
+    post_id.to_string() + "-" + &rarity_level.to_string()
+}
+
+fn is_enough_fees(funds: Vec<Coin>, requireds: Vec<Coin>) -> bool {
+    if requireds.len() == 0 {
+        return true;
+    }
+    // It takes O(n^2) time complexity but both list are extremely small
+    for required in requireds.iter() {
+        if has_coins(&funds, &required) {
+            return true;
+        }
+    }
+    false
+}
+
+fn check_eligibility<'a>(
+    storage: &dyn Storage,
+    querier: &'a dyn Querier,
+    sender: Addr,
+    post_id: u64,
+    engagement_threshold: u32,
+) -> Result<(), ContractError> {
+    let subspace_id = CONFIG.load(storage)?.subspace_id;
+    // Check if the post exists and it is owned by sender.
+    let post = PostsQuerier::new(querier)
+        .query_post(subspace_id, post_id)
+        .map_err(|_| ContractError::PostNotFound { id: post_id })?
+        .post;
+    if post.author != sender {
+        return Err(ContractError::NoEligibilityError {});
+    }
+    // Check if the reactions of the post is larger than the threshold.
+    let reactions = ReactionsQuerier::new(querier).query_reactions(
+        subspace_id,
+        post_id,
+        None,
+        Some(PageRequest {
+            key: None,
+            offset: None,
+            limit: Uint64::new(1),
+            count_total: true,
+            reverse: false,
+        }),
+    )?;
+    let reactions_pagination = reactions.pagination.unwrap_or(PageResponse::default());
+    let reactions_count = reactions_pagination.total.unwrap_or(0u64.into());
+    if engagement_threshold as u64 > reactions_count.into() {
+        return Err(ContractError::NoEligibilityError {});
+    }
+    Ok(())
+}
+
+fn execute_update_admin(
+    deps: DepsMut<DesmosQuery>,
+    info: MessageInfo,
+    new_admin: String,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    check_admin(deps.storage, &info)?;
+    let new_admin_addr = deps.api.addr_validate(&new_admin)?;
+    CONFIG.update(deps.storage, |mut config| -> Result<_, ContractError> {
+        config.admin = new_admin_addr.clone();
```

Comment:
use new
```suggestion
        config.admin = new_admin_addr;
```
Just take the ownership here and use `new_admin` inside the response to avoid a copy

Reply:
Fixed